### PR TITLE
Compatibility with podman v3

### LINKED ISSUE #6446: Check compatibility with podman v3
/kind feature

Check `podman generate kube` vs `podman kube generate`

Blocked on https://github.com/redhat-developer/odo/pull/6385

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind bug

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6446

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Review by valaparthvi — COMMENTED** [review]
Do these changes work with podman v4 though?

**Review by rm3l — COMMENTED** [review]
Now that the "odo-podman-test" GitHub job is [actually passing](https://github.com/redhat-developer/odo/actions/runs/3872851517/jobs/6602188156#step:5:190), I think it is a good time to make the corresponding step mandatory for the job status: 

https://github.com/redhat-developer/odo/blob/32ef20d894ee72814f38f4ab39774008fc68250c/.github/workflows/podman-test.yaml#L24-L26

File: tests/helper/component_podman.go [hunk 1]
```diff
@@ -12,54 +14,72 @@ import (
...
 }
 
 func (o *PodmanComponent) GetEnvVars() map[string]string {
-	podName := fmt.Sprintf("%s-%s", o.name, o.app)
-	podDef := getPodDef(podName)
-	res := map[string]string{}
-	for _, env := range podDef.Spec.Containers[0].Env {
-		res[env.Name] = env.Value
+	envs := o.Exec(o.containerName, "env")
+	fmt.Printf("\n\n%s\n\n", envs)
```
Comment (rm3l) [inline tests/helper/component_podman.go:66]: Maybe we should void printing env vars (for security concerns, if the image contains sensitive data)? Or print to the `GinkgoWriter` to have them printed only if the current test fails (or when running Ginkgo in verbose mode)..

Comment (feloy) [reply tests/helper/component_podman.go:66]: This was added only for debugging. I'll remove the line. Thanks

**Comment by feloy** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests
Flaky test
```
 [FAILED] [17.449 seconds]
odo logs command tests
/go/odo_1/tests/integration/cmd_logs_test.go:14
  when component is created and odo logs is executed
  /go/odo_1/tests/integration/cmd_logs_test.go:69
    when running in Deploy mode
    /go/odo_1/tests/integration/cmd_logs_test.go:129
      [It] should successfully show logs of the running component
      /go/odo_1/tests/integration/cmd_logs_test.go:136
```

**Pull request merged**